packages/flutter_tools/lib/src/platform_plugins.dart

```diff
@@ -227,15 +236,17 @@ class AndroidPlugin extends PluginPlatform implements NativeOrDartPlugin {
 ///   - the [dartPluginClass] that will be the entry point for the plugin's
 ///     Dart code
 /// is required.
-class IOSPlugin extends PluginPlatform implements NativeOrDartPlugin {
+class IOSPlugin extends PluginPlatform implements NativeOrDartPlugin, DarwinPlugin {
   const IOSPlugin({
     required this.name,
     required this.classPrefix,
     this.pluginClass,
     this.dartPluginClass,
     bool? ffiPlugin,
     this.defaultPackage,
-  }) : ffiPlugin = ffiPlugin ?? false;
+    bool? sharedDarwinSource,
+  }) : ffiPlugin = ffiPlugin ?? false,
+       sharedDarwinSource = sharedDarwinSource ?? false;
```

Comment:
Do we actually need to be able to explicitly pass `sharedDarwinSource: null` for some reason? If not we should be able to make it `this.sharedDarwinSource = false` above and not need this I think.

Reply:
It makes it easier to do:
```dart
  sharedDarwinSource: yaml[kSharedDarwinSource] as bool?,
```
You're right the property should be nonnull though even when the constructor allows null.